### ISSUE TITLE
jax.numpy.flip() unable to accept tuple inputs to axis keyword argument

### ISSUE THREAD

**Issue by LouisDesdoigts** [google/jax#6451]
Please:

Hi jax team, it appears that the jax.numpy.flip() function is unable to accept tuple inputs to the axis keyword argument and throws a TypeError when you try to do so. I am currently running jax version 0.2.10. Here is a minimal example:

```python

import jax.numpy as jnp
import numpy as onp

array = jnp.arange(9).reshape([3, 3])
print(array)

# [[0 1 2]
#  [3 4 5]
#  [6 7 8]]

print(onp.flip(array, axis=(0, 1)))

# [[8 7 6]
#  [5 4 3]
#  [2 1 0]]

print(jnp.flip(array, axis=[0, 1]))
```

produces this error:

---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-21-07de0f459e9e> in <module>
      5 print(array)
      6 print(onp.flip(array, axis=(0, 1)))
----> 7 print(jnp.flip(array, axis=[0, 1]))
      8 
      9 # jnp_array_flip = jnp.flip(array, axis=0)

~/anaconda3/envs/main/lib/python3.8/site-packages/jax/_src/numpy/lax_numpy.py in flip(m, axis)
   1079   if axis is None:
   1080     return lax.rev(m, list(range(len(shape(m)))))
-> 1081   return lax.rev(m, [_canonicalize_axis(axis, ndim(m))])
   1082 
   1083 

~/anaconda3/envs/main/lib/python3.8/site-packages/jax/_src/util.py in canonicalize_axis(axis, num_dims)
    273 def canonicalize_axis(axis, num_dims) -> int:
    274   """Canonicalize an axis in [-num_dims, num_dims) to [0, num_dims)."""
--> 275   axis = operator.index(axis)
    276   if not -num_dims <= axis < num_dims:
    277     raise ValueError(

TypeError: 'list' object cannot be interpreted as an integer

Cheers!

**Comment by jakevdp** [comment]
Thanks for the report! I believe this was fixed by #6354, though the fix has not yet made it into a release. Your code produces the expected output if run on the master branch.

**Comment by LouisDesdoigts** [comment]
Fantastic, closing the issue!
